### PR TITLE
chore(internal): remove usage of PY2 and PY3

### PR DESCRIPTION
**Pull request by brettlangdon**
Since we no longer support Python 2.7, we can remove the usage of `ddtrace.internal.compat.PY2` and `ddtrace.internal.compat.PY3`.

**Testing:** There are no public API changes, so we are relying heavily on code review and tests to validate this PR.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by avara1986 — APPROVED** [review]

File: tests/tracer/test_span.py [hunk 1]
```diff
@@ -678,7 +678,3 @@ def get_exception_span(exception):
 
     exception_span = get_exception_span(Exception("DataDog/水"))
     assert "DataDog/水" == exception_span.get_tag(ERROR_MSG)
-
-    if six.PY3:
-        exception_span = get_exception_span(Exception("DataDog/水"))
-        assert "DataDog/水" == exception_span.get_tag(ERROR_MSG)
```
Comment (gnufede) [inline tests/tracer/test_span.py:684]: Not sure of the context, but this block was PY3, not Py2, so it shouldn't be removed right?

File: ddtrace/internal/module.py [hunk 2]
```diff
@@ -122,22 +124,10 @@ def _resolve(path):
...
-
-    def find_loader(fullname):
-        # type: (str) -> Optional[Loader]
-        return getattr(find_spec(fullname), "loader", None)
+def find_loader(fullname):
+    # type: (str) -> Optional[Loader]
+    return getattr(find_spec(fullname), "loader", None)
 
 
 LEGACY_DICT_COPY = sys.version_info < (3, 6)
```
Comment (P403n1x87) [inline ddtrace/internal/module.py:133]: This could be removed too

**Review by P403n1x87 — COMMENTED** [review]
A couple of nits, otherwise LGTM!

**Review by mabdinur — COMMENTED** [review]
This might be beyond the scope of this PR but we can also remove all references to [AsyncioContextProvider](https://github.com/DataDog/dd-trace-py/blob/2757aaefc7f0121e581b3a480dbddf3937e2f024/ddtrace/contrib/asyncio/provider.py#L8) and then deprecate this class.

File: tests/profiling/collector/test_stack.py [hunk 3]
```diff
@@ -345,10 +345,7 @@ def test_new_interval():
   return _f{nump1}()"""
 
 for num in range(MAX_FN_NUM):
-    if six.PY3:
-        exec(FN_TEMPLATE.format(num=num, nump1=num + 1))
-    else:
```
Comment (emmettbutler) [inline tests/profiling/collector/test_stack.py:350]: I wonder what was going on here...

File: tests/tracer/test_span.py (hunk 1 above)
Comment (emmettbutler) [reply tests/tracer/test_span.py:684]: This logic appears to directly duplicate the two lines above it, so I can see how removing it makes sense.

**Review by emmettbutler — APPROVED** [review]
Looks great, thanks for putting some time into cleaning up old stuff.

**Comment by brettlangdon** [comment]
> This might be beyond the scope of this PR but we can also remove all references to [AsyncioContextProvider](https://github.com/DataDog/dd-trace-py/blob/2757aaefc7f0121e581b3a480dbddf3937e2f024/ddtrace/contrib/asyncio/provider.py#L8) and then deprecate this class.

This is a great follow up! Let's try not to bloat this PR too much more.

File: ddtrace/internal/module.py (hunk 2 above)
Comment (brettlangdon) [reply ddtrace/internal/module.py:134]: I'll handle in a follow up. Thanks!

File: tests/tracer/test_span.py (hunk 1 above)
Comment (brettlangdon) [reply tests/tracer/test_span.py:684]: yeah, not sure why the assertions were run twice for Python 3.

**Pull request merged**